linkerd/app/src/env.rs

```diff
@@ -463,10 +467,16 @@ pub fn parse_config<S: Strings>(strings: &S) -> Result<super::Config, EnvError>
         )?
         .unwrap_or(ingress_mode);
 
-        let addr = ListenAddr(
-            outbound_listener_addr?
-                .unwrap_or_else(|| parse_socket_addr(DEFAULT_OUTBOUND_LISTEN_ADDR).unwrap()),
-        );
+        let addr = match outbound_listener_addrs {
+            Ok(Some(addrs)) if addrs.len() == 1 => DualListenAddr(addrs[0], None),
+            Ok(Some(addrs)) if addrs.len() == 2 => DualListenAddr(addrs[0], Some(addrs[1])),
```

Comment:
```suggestion
        let addr = match outbound_listener_addrs? {
            Some(addrs) if addrs.len() == 1 => DualListenAddr(addrs[0], None),
            Some(addrs) if addrs.len() == 2 => DualListenAddr(addrs[0], Some(addrs[1])),
```

I think we ought to fail if parsing errors (otherwise what's the point of introducing the new error variant?)

Reply:
Good catch :+1: